client/jetpack-connect/test/authorize.js

```diff
@@ -304,46 +354,107 @@ describe( 'JetpackAuthorize', () => {
 	} );
 
 	describe( 'getRedirectionTarget', () => {
-		test( 'should redirect to pressable if partnerSlug is "pressable"', () => {
-			const renderableComponent = <JetpackAuthorize { ...DEFAULT_PROPS } />;
-			const component = shallow( renderableComponent );
-			component.setProps( {
-				partnerSlug: 'pressable',
-			} );
-			const target = component.instance().getRedirectionTarget();
+		let originalWindowLocation;
+
+		beforeEach( () => {
+			originalWindowLocation = global.window.location;
+			delete global.window.location;
```

Comment:
I might be missing something, but since `global.window.location` is an object, `originalWindowLocation ` will lead to it, and if we `delete global.window.location`, it will also erase `originalWindowLocation`, won't it?

Also, why do we need to `delete` it if we overwrite it on the next line?

Reply:
> Also, why do we need to delete it if we overwrite it on the next line?

The `location` property is read-only, so we can't replace it with a new one without deleting it first: https://developer.mozilla.org/en-US/docs/Web/API/Window/location.

> I might be missing something, but since global.window.location is an object, originalWindowLocation will lead to it, and if we delete global.window.location, it will also erase originalWindowLocation, won't it?

It's indeed a bit counter-intuitive, but `delete` doesn't delete the referenced object (though it might be released by GC if it's not referenced anywhere else), but the _property_ itself, see: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/delete.

I've recorded two short screencasts for clarity:


https://user-images.githubusercontent.com/81248/196293615-06a5c3f4-964e-4740-93e7-76f6700bc4b0.mp4


https://user-images.githubusercontent.com/81248/196293622-02f935e5-4419-4463-82ef-980f9f5972d1.mp4